src/code/z_player_lib.c

```diff
@@ -1187,21 +1918,275 @@ void Player_DrawCouplesMask(PlayState* play, Player* player) {
+
+Vec3f D_801C0DD8 = { 50.0f, 800.0f, 0.0f };
+Vec3f D_801C0DE4 = { 50.0f, 850.0f, 0.0f };
+Gfx* D_801C0DF0[] = {
+    object_link_goron_DL_010590, object_link_goron_DL_010368, object_link_goron_DL_010140,
+    object_link_goron_DL_00FF18, object_link_goron_DL_00FCF0,
+};
+
+Vec2f D_801C0E04[PLAYER_FORM_MAX] = {
+    { 140.0f, -130.0f }, { 0.0f, -200.0f }, { -160.0f, 0.0f }, { 220.0f, -200.0f }, { 0.0f, 0.0f },
+};
+
+Gfx* D_801C0E2C[] = {
+    object_link_nuts_DL_007A28, object_link_nuts_DL_0077D0, object_link_nuts_DL_007548,
+    object_link_nuts_DL_007900, object_link_nuts_DL_0076A0,
+};
+Vec3f D_801C0E40[PLAYER_FORM_MAX] = {
+    { 0.0f, 0.0f, 0.0f },       { -578.3f, -1100.9f, 0.0f }, { -189.5f, -594.87f, 0.0f },
+    { -570.0f, -812.0f, 0.0f }, { -230.0f, -520.0f, 0.0f },
+};
+Vec3f D_801C0E7C = { 1100.0f, -700.0f, 0.0f };
+
+// unused
+Vec3f D_801C0E88 = { 1600.0f, -1700.0f, -70.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_801278F8.s")
+Vec3f D_801C0E94 = { 1800.0f, -300.0f, 0.0f };
+Vec3f D_801C0EA0 = { 1300.0f, -400.0f, 0.0f };
+Vec3f D_801C0EAC = { 630.0f, 100.0f, -30.0f };
+Vec3s D_801C0EB8 = { 0, 0, 0x7FFF };
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80127A60.s")
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80127B64.s")
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80127BE8.s")
 
+void func_80127DA4(PlayState* play, struct_801F58B0 arg1[], struct_80128388_arg1 arg2[], s32 arg3, Vec3f* arg4,
+                   Vec3f* arg5, u32* arg6);
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80127DA4.s")
 
+void func_80128388(struct_801F58B0 arg0[], struct_80128388_arg1 arg1[], s32 arg2, Mtx** arg3);
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80128388.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_801284A0.s")
+void func_801284A0(PlayState* play, Player* player) {
+    s32 pad;
+    Mtx* sp90 = GRAPH_ALLOC(play->state.gfxCtx, 6 * sizeof(Mtx));
+    Vec3f sp84;
+    Vec3f sp78;
+    Vec3f* iter = D_801C0C0C;
+    Vec3f* iter2 = D_801C0C30;
+    u32 sp6C = play->gameplayFrames;
+    s32 i;
+
+    OPEN_DISPS(play->state.gfxCtx);
+
+    gSPSegment(POLY_OPA_DISP++, 0x0B, sp90);
+
+    Matrix_MultVec3f(&D_801C0C00, &D_801C0C54[1].unk_08);
+    Math_Vec3f_Lerp(&player->bodyPartsPos[7], &player->bodyPartsPos[0], 0.2f, &D_801C0C54[2].unk_08);
+
+    for (i = 0; i < ARRAY_COUNT(D_801C0C0C); i++) {
+        Matrix_MultVec3f(iter, &sp84);
+        Matrix_MultVec3f(iter2, &sp78);
+
+        func_80127DA4(play, D_801F58B0[i], D_801C0C54, 3, &sp84, &sp78, &sp6C);
+        sp6C += 11;
+
+        Matrix_Push();
+        Matrix_Translate(iter->x, iter->y, iter->z, MTXMODE_APPLY);
+        func_80128388(D_801F58B0[i], D_801C0C54, 3, &sp90);
+        Matrix_Pop();
+        iter++;
+        iter2++;
+    }
+
+    CLOSE_DISPS(play->state.gfxCtx);
+}
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80128640.s")
 
```

Comment:
looking at m2c output, `func_*B74` looks tiny, is it being reserved for another PR for some reason?

```c
void func_80128B74(s16 **arg0, s16 *arg1, s16 **arg2) {
    void *temp_a3;

    temp_a3 = (arg1->unk14B * 0xC) + &D_801C0D24;
    Actor_SetFeetPos(arg1, arg2, 9, temp_a3, 6, temp_a3);
}
```